core/src/main/java/io/substrait/extendedexpression/ExtendedExpressionProtoConverter.java

```diff
@@ -0,0 +1,60 @@
+package io.substrait.extendedexpression;
+
+import io.substrait.expression.Expression;
+import io.substrait.expression.proto.ExpressionProtoConverter;
+import io.substrait.extension.ExtensionCollector;
+import io.substrait.proto.AggregateFunction;
+import io.substrait.proto.ExpressionReference;
+import io.substrait.proto.ExtendedExpression;
+import io.substrait.type.proto.TypeProtoConverter;
+
+/** Converts from {@link ExtendedExpression} to {@link ExtendedExpression} */
+public class ExtendedExpressionProtoConverter {
+  public ExtendedExpression toProto(
+      io.substrait.extendedexpression.ExtendedExpression extendedExpression) {
+
+    ExtendedExpression.Builder builder = ExtendedExpression.newBuilder();
+    ExtensionCollector functionCollector = new ExtensionCollector();
+
+    final ExpressionProtoConverter expressionProtoConverter =
+        new ExpressionProtoConverter(functionCollector, null);
+
+    for (io.substrait.extendedexpression.ExtendedExpression.ExpressionReference
+        expressionReference : extendedExpression.getReferredExpressions()) {
+      io.substrait.extendedexpression.ExtendedExpression.ExpressionTypeReference
+          expressionTypeReference = expressionReference.getExpressionType();
+      if (expressionTypeReference
+          instanceof io.substrait.extendedexpression.ExtendedExpression.ExpressionType et) {
```

Comment:
I made a small change here (https://github.com/substrait-io/substrait-java/pull/206/commits/92d2cc57e46e4b4972e5251477abf6d9baf4300b) to bind the variables your instanceof checking to make the PR easier to read. Using this, this way you can avoid the need to recast them.